### PR TITLE
add stripe modules and configuration for external checkout

### PR DESCRIPTION
**Pull request by jopieji**
Added stripe external checkout

Private key might need to be manually configured for testing; have had trouble with @Value()

The checkout doesn't compute prices of cart items yet, and all items need to be manually programmed into stripe (at least for our test flow)

### CODE REVIEW

File: src/main/java/com/greengroupecommerce/controllers/StripeController.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+public class StripeController {
+
+    @PostMapping
+    public RedirectView createCheckoutSession() throws Exception {
+        @Value("${stripe.api.secretKey}")
+        final String secretKey;
+
+        Stripe.apiKey = secretKey;
+
+        String YOUR_DOMAIN = "http://localhost:8080";
```
Comment (ZahraAstrom) [inline src/main/java/com/greengroupecommerce/controllers/StripeController.java:25]: I noticed you mapped this to the dev port, is it gonna be an issue in prod?

File: src/main/java/com/greengroupecommerce/controllers/StripeController.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        SessionCreateParams params =
+                SessionCreateParams.builder()
+                        .setMode(SessionCreateParams.Mode.PAYMENT)
+                        .setSuccessUrl(YOUR_DOMAIN + "/confirmation")
+                        .setCancelUrl(YOUR_DOMAIN + "/home")
+                        .addLineItem(
+                                SessionCreateParams.LineItem.builder()
+                                        .setQuantity(1L)
+                                        // Provide the exact Price ID (for example, pr_1234) of the product you want to sell
+                                        .setPrice("price_1O2Po6AJzwVAVVDsohgTU5z1")
```
Comment (ZahraAstrom) [inline src/main/java/com/greengroupecommerce/controllers/StripeController.java:35]: should we tweak the Product table schema to have a column that stores these price IDs?

**Review by ZahraAstrom — APPROVED** [review]

File: src/main/java/com/greengroupecommerce/controllers/StripeController.java (hunk 1 above)
Comment (jopieji) [reply src/main/java/com/greengroupecommerce/controllers/StripeController.java:25]: Yes, forgot to change.

File: src/main/java/com/greengroupecommerce/controllers/StripeController.java (hunk 2 above)
Comment (jopieji) [reply src/main/java/com/greengroupecommerce/controllers/StripeController.java:35]: Most likely. Wasn't going to increment the database version until we had all the products we want added to stripe.

**Pull request merged**